lib/wallet/txdb.js

```diff
@@ -845,6 +845,36 @@ class TXDB {
     return false;
   }
 
+  /**
+   * Test whether the covenant
+   * has a duplicate open.
+   * @param {Covenant}
+   * @returns {Boolean}
+   */
+
+  async isCovenantDoubleOpen(covenant) {
```

Comment:
Few comments:

1. It seems like this is copied from `isDoubleOpen(tx)`, correct? Could you rewrite `isDoubleOpen(tx)` so that it uses this `isCovenantDoubleOpen(covenant)` function?
2. Reading this code actually makes me think that there is a bug with the original `isDoubleOpen` that you copied it from! In the original, we `return false` from inside the for loop if the duplicate open is from an old auction. But that doesn't mean the whole transaction doesn't have a double open, just that covenant isn't! It should `continue` and not `return false`.

We can rewrite `isDoubleOpen` as something like:
```javascript
async isDoubleOpen(tx) {
  const isDoubleOpenList = await Promise.all(
    tx.outputs.map(({ covenant }) => isCovenantDoubleOpen(covenant))
  );
  // any covenant being a double open means the whole transaction includes a double open
  return isDoubleOpenList.some(Boolean);
}
```

Which fixes the `continue` bug.

3. Reviewing this code also caused me to review `removeDoubleOpen(tx)`. Handshake used to have a bug that prevented valid double opens (when auction expires). `isDoubleOpen` fixed that bug, but `removeDoubleOpen` never did! It seems you're not using this `removeDoubleOpen` function anyway so no change necessary in this PR, just pointing it out.

Reply:
removeDoubleOpen will keep it in mind, i am refactoring as you have indicated. I didnt realize the "bug" you mentioned above tough, addressing that one as well.